### PR TITLE
Generalize taylor `GetInitLogDensity` for factorial models

### PR DESCRIPTION
**Pull request by Sahel13**
The change made in #252 makes the assumption that the prior distribution for all factors is the same. This PR removes that.

### CODE REVIEW

File: cuthbert/gaussian/taylor/non_associative_filter.py [hunk 1]
```diff
@@ -89,23 +90,23 @@ def init_prepare(
     model_inputs = tree.map(lambda x: jnp.asarray(x), model_inputs)
     init_log_density, linearization_point = get_init_log_density(model_inputs)
 
-    # Handle factorial axis if present
-    lin_point_dim = linearization_point.ndim
-    linearization_point = jnp.atleast_2d(linearization_point)
+    linearization_shape = linearization_point.shape
```
Comment (SamDuffield) [inline cuthbert/gaussian/taylor/non_associative_filter.py:94]: Can we just add a comment in here to make it clear that the reshape code is to support a factorial axis? Otherwise a newcomer looking at the code would be quite confused 😄

**Review by SamDuffield — COMMENTED** [review]
This is looks great, just a small comment useful for our future selves haha

File: cuthbert/gaussian/taylor/non_associative_filter.py (hunk 1 above)
Comment (Sahel13) [reply cuthbert/gaussian/taylor/non_associative_filter.py:94]: Absolutely

**Review by SamDuffield — APPROVED** [review]
